### PR TITLE
Port tests to xUnit.net 2

### PR DESCRIPTION
**Pull request by bricelam**
This will enable them to cross-target .NET Core 3.0

### CODE REVIEW

File: test/EntityFramework/FunctionalTests/ProductivityApi/SimpleScenarios.cs [hunk 1]
```diff
@@ -15,7 +15,6 @@ namespace ProductivityApiTests
     ///     They are here as a good starting point for understanding the functionality and to give some level
     ///     of confidence that the simple scenarios keep on working as the code evolves.
     /// </summary>
-    [PartialTrustFixture]
```
Comment (bricelam) [inline test/EntityFramework/FunctionalTests/ProductivityApi/SimpleScenarios.cs]: @ajcvickers Is it OK to remove the partial trust tests or should I port them too?

Comment (ajcvickers) [reply test/EntityFramework/FunctionalTests/ProductivityApi/SimpleScenarios.cs]: Remove them!

**Comment by bricelam** [comment]
lol, I knew it would deadlock with all those Wait and Result calls.

**Comment by roji** [comment]
Ah yes, xunit's deadlock detection feature :) https://github.com/xunit/xunit/issues/864

**Comment by roji** [comment]
FYI just if it's too much work to fix the sync-over-async situation, you may be able to work around the issue via `-maxthreads unlimited`.

**Pull request merged**